src/resource_clients/request_queue.ts

```diff
@@ -190,9 +192,12 @@ export class RequestQueueClient extends ResourceClient {
                 break;
             }
 
-            // Sleep for some time before trying again
-            // TODO: Do not use delay from client, but rather a separate value
-            await new Promise((resolve) => setTimeout(resolve, this.httpClient.minDelayBetweenRetriesMillis));
+            // Exponential backoff
+            const delayMillis = Math.max(
+                Math.floor(Math.random() * (2 ** i)) * minDelayBetweenUnprocessedRequestsRetriesMillis,
+                minDelayBetweenUnprocessedRequestsRetriesMillis,
+            );
```

Comment:
I think if you have the random factor between 0 and 1, you'll often run into situations where the delay is shorter than it should. By having the random factor between 1 and 2, you can simplify the code a bit, and have more consistently increasing delays:
```suggestion
            const delayMillis = Math.floor((1 + Math.random()) * (2 ** i) * minDelayBetweenUnprocessedRequestsRetriesMillis);
```

Reply:
I copied it from apify-core batch rq, so I guess it's also wrong there

Reply:
I guess it's relevant even for this https://github.com/apify/apify-client-js/pull/243#discussion_r866949635